BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
+        var response = JsonSerializer.Deserialize<CnbRatesResponse>(data);
```

Comment:
The deserialization and filtering logic could benefit from being split into separate methods. We want everything to do only what it absolutely has to, everything else is separated when possible.

For example:
```C#
private CnbRatesResponse DeserializeRates(string data)
{
    return JsonSerializer.Deserialize<CnbRatesResponse>(data)
           ?? throw new InvalidOperationException("Failed to deserialize rates.");
}

private IEnumerable<Rate> FilterRates(CnbRatesResponse response, List<string> requestedCurrencies)
{
    return response.Rates.Where(rate => requestedCurrencies.Contains(rate.CurrencyCode));
}
```

With these methods, the main function becomes much cleaner:
```C#
var data = await FetchRatesAsync();
var response = DeserializeRates(data); 
var filteredRates = FilterRates(response, currencies);
```
We can then add a test for these two functions separately 🙂 

This approach makes each step of the process easier to understand and test independently. It also allows for better error handling—if deserialization or filtering fails, you know exactly where the problem occurred.